skimage/_shared/tests/test_safe_as_int.py

```diff
@@ -1,21 +1,31 @@
 import numpy as np
+import pytest
 from skimage._shared.utils import safe_as_int
 
 
 def test_int_cast_not_possible():
-    np.testing.assert_raises(ValueError, safe_as_int, 7.1)
-    np.testing.assert_raises(ValueError, safe_as_int, [7.1, 0.9])
-    np.testing.assert_raises(ValueError, safe_as_int, np.r_[7.1, 0.9])
-    np.testing.assert_raises(ValueError, safe_as_int, (7.1, 0.9))
-    np.testing.assert_raises(ValueError, safe_as_int, ((3,   4,   1),
-                                                       (2, 7.6, 289)))
-
-    np.testing.assert_raises(ValueError, safe_as_int, 7.1, 0.09)
-    np.testing.assert_raises(ValueError, safe_as_int, [7.1, 0.9], 0.09)
-    np.testing.assert_raises(ValueError, safe_as_int, np.r_[7.1, 0.9], 0.09)
-    np.testing.assert_raises(ValueError, safe_as_int, (7.1, 0.9), 0.09)
-    np.testing.assert_raises(ValueError, safe_as_int, ((3,   4,   1),
-                                                       (2, 7.6, 289)), 0.25)
+    with pytest.raises(ValueError):
```

Comment:
Parametrized tests make the tests really much more complicated to read. Matplotlib seems to be going into that direction (though there is still a debate), but I find it makes the code very hard to maintain. I think here you could do a simple loop, and it would work fine.

I haven't seen the class-based approach used much in the scientific python community. `scikit-image` does have a few, and it might be a good idea at some point to choose one and only one way to write tests.